From: su henry <henry.su.ati@gmail.com>
Date: Fri, 20 Jul 2007 12:07:46 +0000 (-0400)
Subject: The SATA controller device ID is different according to
X-Git-Tag: v2.6.23-rc1~219^2~7
X-Git-Url: http://git.kernel.org/?p=linux%2Fkernel%2Fgit%2Ftorvalds%2Flinux-2.6.git;a=commitdiff_plain;h=badc2341579511a247f5993865aa68379e283c5c

The SATA controller device ID is different according to
the onchip SATA type set in the system BIOS:
Device                              Device ID
SATA in IDE mode             0x4390
SATA in AHCI mode           0x4391
SATA in non-raid5 driver     0x4392
SATA in raid5 driver           0x4393

Although the device ID is different, they use the same AHCI driver
.The attached file is the patch for adding these device
IDs for ATI SB700.

Signed-off-by: henry.su.ati@gmail.com
Signed-off-by: Jeff Garzik <jeff@garzik.org>
---

Backported to Debian's 2.6.18 by dann frazier <dannf@debian.org>

diff -urpN linux-source-2.6.18.orig/drivers/scsi/ahci.c linux-source-2.6.18/drivers/scsi/ahci.c
--- linux-source-2.6.18.orig/drivers/scsi/ahci.c	2007-08-03 01:41:46.646787000 -0600
+++ linux-source-2.6.18/drivers/scsi/ahci.c	2007-08-03 02:08:45.170323428 -0600
@@ -348,7 +348,13 @@ static const struct pci_device_id ahci_p
 	{ PCI_VENDOR_ID_ATI, 0x4381, PCI_ANY_ID, PCI_ANY_ID, 0, 0,
 	  board_ahci }, /* ATI SB600 raid */
 	{ PCI_VENDOR_ID_ATI, 0x4390, PCI_ANY_ID, PCI_ANY_ID, 0, 0,
-	  board_ahci }, /* ATI SB700 */
+	  board_ahci }, /* ATI SB700 IDE */
+	{ PCI_VENDOR_ID_ATI, 0x4391, PCI_ANY_ID, PCI_ANY_ID, 0, 0,
+	  board_ahci }, /* ATI SB700 AHCI */
+	{ PCI_VENDOR_ID_ATI, 0x4392, PCI_ANY_ID, PCI_ANY_ID, 0, 0,
+	  board_ahci }, /* ATI SB700 nraid5 */
+	{ PCI_VENDOR_ID_ATI, 0x4393, PCI_ANY_ID, PCI_ANY_ID, 0, 0,
+	  board_ahci }, /* ATI SB700 raid5 */
 
 	/* VIA */
 	{ PCI_VENDOR_ID_VIA, 0x3349, PCI_ANY_ID, PCI_ANY_ID, 0, 0,
